### PR TITLE
Rework argument parsing.

### PR DESCRIPTION
**Pull request by zherczeg**
Heavily simplify the interface of CLI.

### CODE REVIEW

File: jerry-main/main-unix.c [hunk 1]
```diff
@@ -513,25 +512,29 @@ main (int argc,
...
-        check_usage (false, argv[0], "Error: unrecognized option: ", cli_state.arg[0]);
+        cli_state.error = "Internal error";
         break;
       }
     }
   }
 
+  if (cli_state.error != NULL)
+  {
+    if (cli_state.arg != NULL)
```
Comment (LaszloLango) [inline jerry-main/main-unix.c]: Missing braces

File: jerry-main/cli.c [hunk 2]
```diff
@@ -37,173 +37,233 @@
...
+    .opts = options
+  };
+} /* cli_init */
 
 /**
- * Print the prefix of a string.
+ * Checks whether the current argument is an option.
+ *
+ * Note:
+ *   The state->error is not NULL on error and it contans the error message.
```
Comment (LaszloLango) [inline jerry-main/cli.c]: typo: `contains`

File: jerry-main/cli.c [hunk 3]
```diff
@@ -37,173 +37,233 @@
...
 /**
- * Perform one step of the command line option processor.
+ * Returns the next argument as integer.
  *
- * @return the ID of the option that was found. (The definition of the found
- *         option (if any) is available via state->opt, while the option
- *         string and its arguments are available via state->arg[0..].)
- *         CLI_OPT_END signals that all command line arguments are consumed.
+ * Note:
+ *   The state->error is not NULL on error and it contans the error message.
```
Comment (LaszloLango) [inline jerry-main/cli.c]: ditto

**Comment by zherczeg** [comment]
Thanks, fixed.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**